### PR TITLE
Assignee selector

### PR DESCRIPTION
**Pull request by MargaretKrutikova**

### CODE REVIEW

File: src/components/Dropdown.re [hunk 1]
```diff
@@ -0,0 +1,170 @@
+open ClickOutside;
+
+let str = ReasonReact.string;
```
Comment (Yakimych) [inline src/components/Dropdown.re:3]: We should probably define it in some globally accessible module and use it from everywhere.

File: src/components/Dropdown.re [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+let make =
+    (
+      ~selected: option(string),
+      ~options: array(string),
+      ~selectOption: string => unit,
+      ~renderDropdownButton=?,
+      ~align=Center,
+    ) => {
+  let (state, dispatch) = React.useReducer(reducer, {isOpen: false});
+  let select = (item: string) => {
```
Comment (Yakimych) [inline src/components/Dropdown.re]: `selectAndClose`?

File: src/components/Dropdown.re [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+             {options->Belt.Array.mapWithIndex((index, item) =>
+                <div
+                  key={string_of_int(index)}
+                  onClick={_ => select(item)}
+                  className={Styles.optionItem(
+                    selected->Belt.Option.mapWithDefault(false, val_ =>
+                      val_ === item
+                    ),
+                  )}>
+                  {item |> ReasonReact.string}
```
Comment (Yakimych) [inline src/components/Dropdown.re]: `item |> str`?

File: src/components/Grid.re [hunk 4]
```diff
@@ -1,32 +1,83 @@
...
-      @ [media(Breakpoints.up(Md), gridItem(Theme.spacingPx.medium))]
+module Styles = {
+  let containerSpacing = (spacing: int) => [
+    width(`calc((`add, pct(100.0), px(spacing)))),
+    margin(px(- spacing / 2)),
+  ];
+  let itemSpacing = (spacing: int) => [padding(px(spacing / 2))];
+
+  let calculateColumnsWidth = (columns: int) =>
+    float_of_int(columns) /. 12.0 *. 100.0;
```
Comment (Yakimych) [inline src/components/Grid.re:33]: What is `/.` and `*.`? What does the dot do?

File: src/components/Grid.re [hunk 5]
```diff
@@ -1,32 +1,83 @@
...
   };
-  <div className={Css.style(getGridStyles())}> children </div>;
+};
+
+[@react.component]
+let make = (~type_: gridType, ~children, ~className=?) => {
+  let gridClassName = getGridStyles(type_);
+  let customClassName = Belt.Option.getWithDefault(className, "");
+
+  <div className={j|$gridClassName $customClassName|j}> children </div>;
```
Comment (Yakimych) [inline src/components/Grid.re:82]: What's happening here? Is it JS interop? Why is it necessary?

File: src/components/ShoppingItem.re [hunk 6]
```diff
@@ -10,20 +37,48 @@ let make = (~item: shoppingItem, ~onItemChange, ~id: int) => {
...
+        selectOption=handleAssigneeChange
+        renderDropdownButton={toggle =>
+          <button onClick=toggle className=Styles.assigneeBtn>
+            {switch (item.assignee) {
+             | None => ReasonReact.string("+")
+             | Some(value) =>
+               value
+               |> getAssigneeLetter
+               |> String.uppercase
+               |> ReasonReact.string
```
Comment (Yakimych) [inline src/components/ShoppingItem.re:78]: `|> str`?

File: src/hooks/ClickOutside.re [hunk 7]
```diff
@@ -0,0 +1,30 @@
+open Webapi.Dom;
+
+let handleClickOutside = (domElement: Dom.element, e: Dom.mouseEvent, fn) => {
+  let targetElement = MouseEvent.target(e) |> EventTarget.unsafeAsElement;
+
+  !(domElement |> Element.contains(targetElement)) ? fn(e) : ();
+};
+
+let useClickOutside = (onClickOutside: Dom.mouseEvent => unit) => {
```
Comment (Yakimych) [inline src/hooks/ClickOutside.re:9]: Is this a custom hook?

**Review by Yakimych — CHANGES_REQUESTED** [review]

File: src/hooks/ClickOutside.re (hunk 7 above)
Comment (MargaretKrutikova) [reply src/hooks/ClickOutside.re:9]: Yes :)

File: src/components/Grid.re (hunk 4 above)
Comment (MargaretKrutikova) [reply src/components/Grid.re:33]: It is for `float` types.

File: src/components/Grid.re (hunk 5 above)
Comment (MargaretKrutikova) [reply src/components/Grid.re:82]: It is a string interpolation.

**Pull request merged**